common/error_handler.go

```diff
@@ -13,10 +14,26 @@ const (
 	unknownError          = "Unknown Error"
 )
 
-func HandleError(ctx context.Context, err error) HTTPError {
+func HandleError(ctx context.Context, w http.ResponseWriter, kind Kind, message string, cause error, httpErrorMapper func(context.Context, error) *HTTPError) {
+	err := CreateError(ctx, kind, message, cause)
 	logEntry := GetLogEntryFromContext(ctx)
 	logEntry.Error(err)
+	httpError := httpErrorMapper(ctx, err)
 
+	if httpError == nil {
+		switch t := cause.(type) {
+		case CustomError:
+			e := t.HTTPError(ctx)
+			httpError = &e
+		default:
+			e := MapError(ctx, err)
+			httpError = &e
```

Comment:
Can this be a single line?

Reply:
> Can this be a single line?

No unfortunately. `&MapError(ctx, err)` is invalid Go.